velox/connectors/hive/tests/PartitionIdGeneratorTest.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/PartitionIdGenerator.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+#include "gtest/gtest.h"
+
+namespace facebook::velox::connector::hive {
+
+class PartitionIdGeneratorTest : public ::testing::Test,
+                                 public test::VectorTestBase {};
+
+TEST_F(PartitionIdGeneratorTest, consecutiveIds) {
+  vector_size_t numPartitions = 100;
+
+  PartitionIdGenerator idGenerator(ROW({VARCHAR()}), {0}, 100);
+
+  RowVectorPtr input = makeRowVector(
+      {makeFlatVector<StringView>(numPartitions * 3, [&](auto row) {
+        return StringView(Date(18000 + row % numPartitions).toString());
+      })});
+
+  raw_vector<uint64_t> ids;
+  idGenerator.run(input, ids);
+
+  // distinctIds contains ids in [1, 100]
+  std::unordered_set<uint64_t> distinctIds(ids.begin(), ids.end());
+  EXPECT_EQ(*std::min_element(distinctIds.begin(), distinctIds.end()), 1);
+  EXPECT_EQ(
+      *std::max_element(distinctIds.begin(), distinctIds.end()), numPartitions);
+  EXPECT_EQ(distinctIds.size(), numPartitions);
+}
+
+TEST_F(PartitionIdGeneratorTest, stableIds) {
+  vector_size_t numPartitions = 50;
```

Comment:
This variable name is confusing because the code creates 100 partitions, not 50.

Reply:
Revised